gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/FlowExecutionResourceHandler.java

```diff
@@ -36,6 +36,14 @@
    */
   public List<FlowExecution> getLatestFlowExecution(PagingContext context, FlowId flowId, Integer count, String tag, String executionStatus);
 
+  /**
+   * Get latest {@link FlowExecution} for every flow in `flowGroup`
+   *
+   * NOTE: `executionStatus` param not provided yet, without justifying use case, due to complexity of interaction with `count`
+   * and resulting efficiency concern of performing across many flows sharing the single named group.
+   */
+  public List<FlowExecution> getLatestFlowGroupExecutions(PagingContext context, String flowGroup, Integer count, String tag);
```

Comment:
this count is semantically the maximum number of executions for each flow found within the group, not the total number of `FlowExecution`s to return.  that said, following the lead of `getLatestFlowExecution` (directly above), which also ignores it, I never inspect or utilize the `PagingContext`.  I presumed it to be a mandated param, but if it would be legal to omit it, that would be a clearer statement of intent (to ignore).  do you know if I could?

Reply:
How many results can we return in this response? There could be 100k flows, and if we return all of them in one response, it might take too long to generate it and the request will time out.

If this count is not about the number of flowExecutions, then can we name it differently to make this more clear? Like "maxExecutionsPerFlow" or something like that?

Reply:
as we agreed offline, I've renamed `count` to `countPerFlow` (on the `latestFlowGroupExecutions` finder).

AFA potential to time out from a large number of historical flow executions (within the group), there is the `DatasetCleaner`, scheduled by `KafkaJobStatusMonitor` to run in the background.  that can constrain historical flow executions from accumulating forever.  when the config property named by `MultiCleanableDatasetFinder.datasetFinderClassKey` is set to
`org.apache.gobblin.data.management.retention.dataset.finder.TimeBasedDatasetStoreDatasetFinder`, the latter will scan the `MysqlJobStatusStateStore` to offer the contents up for retention/expiration.

so while there could still be lots of prior executions, there are constraints on accumulation.  I propose we gain operational experience to determine whether there's a need to truncate or page through the results: in practice we may not be in danger of timing out (but I agree with you that it's worth keeping an eye out for).